### PR TITLE
Fix multiple threads do vfork

### PR DESCRIPTION
**Pull request by jessehui**

### CODE REVIEW

File: src/libos/src/process/do_vfork.rs [hunk 1]
```diff
@@ -84,21 +112,53 @@ pub fn do_vfork(mut context: *mut CpuContext) -> Result<isize> {
         mem::replace(&mut *current_file_table, new_file_table)
     };
     if let Some(_) = vfork_file_tables.insert(parent_pid, parent_file_table) {
```
Comment (volcano0dr) [inline src/libos/src/process/do_vfork.rs:114]: I think it makes more sense to use `try_insert`.

Comment (jessehui) [reply src/libos/src/process/do_vfork.rs]: Rewrite the logic and put file table saving after stop all other threads.

File: src/libos/src/process/thread/mod.rs [hunk 2]
```diff
@@ -304,10 +305,17 @@ impl Thread {
...
     pub fn force_stop(&self) {
         let mut inner = self.inner();
-        inner.stop();
+        // If the thread is not exited or stopped, then notify it to stop
+        if inner.status() != ThreadStatus::Exited && inner.status() != ThreadStatus::Stopped {
+            inner.notify_stop();
+        }
     }
 
     pub fn is_forced_to_stop(&self) -> bool {
```
[the remark this answers is not in the document]
Comment (jessehui) [reply src/libos/src/process/thread/mod.rs:314]: Refine state check when calling `is_forced_to_stop`. Thanks for your review.

File: src/libos/src/process/do_vfork.rs [hunk 3]
```diff
@@ -101,6 +99,48 @@ pub fn is_vforked_child_process() -> bool {
...
+    let ret = vfork_file_tables.insert(parent_pid, parent_file_table);
+    debug_assert!(ret.is_none());
+    Ok(())
+}
+
+fn vfork_stop_all_child_thread(current: &ThreadRef) {
+    // stop all other child threads
+    loop {
+        let child_threads = current.process().threads();
+        let running_threads = child_threads
```
Comment (volcano0dr) [inline src/libos/src/process/do_vfork.rs:123]: `let running_threads_count = child_threads.iter().filter(...).map(|thread| ....).count()`

Comment (jessehui) [reply src/libos/src/process/do_vfork.rs]: Done. Thanks for your review.

**Review by volcano0dr — APPROVED** [review]
LGTM

**Pull request merged**